File: benchmarks/record_build.go
package main

import (
	"fmt"
	"time"
)

type Stack struct {
	items []int
	total int
}

func build(n int) Stack {
	s := Stack{items: []int{}, total: 0}
	i := 0
	for i < n {
		s.items = append(s.items, i)
		s.total = s.total + i
		i = i + 1
	}
	return s
}

func main() {
	t0 := time.Now()
	s := build(500000)
	fmt.Println(s.total + len(s.items))
	fmt.Printf("bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/record_build.rs
use std::time::Instant;

struct Stack {
    items: Vec<i32>,
    total: i32,
}

fn build(n: i32) -> Stack {
    let mut s = Stack { items: Vec::new(), total: 0 };
    let mut i = 0;
    while i < n {
        s.items.push(i);
        s.total += i;
        i += 1;
    }
    s
}

fn main() {
    let t0 = Instant::now();
    let s = build(500000);
    println!("{}", s.total + s.items.len() as i32);
    println!("bench_ns={}", t0.elapsed().as_nanos());
}
